From 6a9a7d0dbe5dc47406a9767e319633f13c6bdbe6 Mon Sep 17 00:00:00 2001
From: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
Date: Wed, 6 May 2020 17:55:32 +0530
Subject: [PATCH 1357/1851] mtd: cadence-quadspi: Added support for is25wx256

commit 3ff8d4efec3590e44ca8c59106739caf48609aeb from
https://github.com/Xilinx/linux-xlnx.git

This patch adds support for OSPI ISSI part(is25wx256).

Signed-off-by: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index d7dc7fcd281b..5c00da8c6163 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -2595,6 +2595,9 @@ static const struct flash_info spi_nor_ids[] = {
 			SECT_4K | SPI_NOR_DUAL_READ |
 			SPI_NOR_QUAD_READ | SPI_NOR_HAS_LOCK |
 			SPI_NOR_4B_OPCODES) },
+	{ "is25wx256",  INFO(0x9d5b19, 0, 128 * 1024, 256,
+			     SECT_4K | USE_FSR | SPI_NOR_OCTAL_READ |
+			     SPI_NOR_OCTAL_WRITE | SPI_NOR_4B_OPCODES) },
 
 	/* Macronix */
 	{ "mx25l512e",   INFO(0xc22010, 0, 64 * 1024,   1, SECT_4K) },
@@ -5531,7 +5534,8 @@ int spi_nor_scan(struct spi_nor *nor, const char *name,
 		nor->flags |= SNOR_F_HAS_LOCK;
 
 	if (((u16)JEDEC_MFR(nor->info) != SNOR_MFR_MICRON) &&
-	    ((u16)JEDEC_MFR(nor->info) != SNOR_MFR_CYPRESS))
+	    ((u16)JEDEC_MFR(nor->info) != SNOR_MFR_CYPRESS) &&
+	    ((u16)JEDEC_MFR(nor->info) != SNOR_MFR_ISSI))
 		nor->flags |= SNOR_F_BROKEN_OCTAL_DDR;
 
 	/*
-- 
2.31.1

